### PR TITLE
Command APIs of ThingIFAPI

### PR DESCRIPTION
**Pull request by yongpingchen**
- introduce simple-mock for mocking classes. Need to update npm packages and typings
- implement ThingIFAPI#postNewCommand, getCommand, listCommands
- small tests and large tests for these new APIs

### CODE REVIEW

**Comment by yongpingchen** [comment]
@nfukuzaki can you review it?

File: test/large/ThingIFAPICommandTest.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            expect(cmd.commandID).not.equal("");
+            return thingIFAPI.postNewCommand(postCommandRequest);
+        }).then((cmd)=>{
+            expect(cmd).not.null;
+            expect(cmd.commandID).not.equal("");
+            return thingIFAPI.getCommand(cmd.commandID);
+        }).then(()=>{
+            return thingIFAPI.listCommands()
+        }).then((result)=>{
+            expect(result.results.length).to.equal(2);
```
Comment (nfukuzaki) [inline test/large/ThingIFAPICommandTest.ts]: please verify command instance

File: test/large/ThingIFAPICommandTest.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        thingIFAPI.postNewCommand(postCommandRequest).then((cmd)=>{
+            expect(cmd).not.null;
+            expect(cmd.commandID).not.equal("");
+            return thingIFAPI.postNewCommand(postCommandRequest);
+        }).then((cmd)=>{
+            expect(cmd).not.null;
+            expect(cmd.commandID).not.equal("");
+            return thingIFAPI.getCommand(cmd.commandID);
+        }).then(()=>{
+            return thingIFAPI.listCommands()
```
Comment (nfukuzaki) [inline test/large/ThingIFAPICommandTest.ts]: please verify command instance.

**Comment by yongpingchen** [comment]
Improved. Please check

**Pull request merged**